agent/agents/supervisor/supervisor_test.go

```diff
@@ -259,6 +259,44 @@ func TestSupervisor(t *testing.T) {
 	})
 }
 
+func TestStartProcessFail(t *testing.T) {
```

Comment:
can we include this test to test group above? Seems like it supposed to be there

Reply:
TestSupervisor is a function with a group of tests where the states of different processes are highly interconnected, making it challenging to add a new individual test seamlessly. Personally, I find that using a new context would offer a cleaner and more straightforward way to verify this case.